tests/components/cast/test_media_player.py

```diff
@@ -358,78 +387,429 @@ async def test_update_cast_chromecasts(hass):
+    assert "Invalid JSON in media_content_id" in caplog.text
+    chromecast.start_app.assert_not_called()
+    quick_play_mock.assert_not_called()
+
+    # Play_media - media_type cast with extra keys
+    await common.async_play_media(
+        hass, "cast", '{"app_id": "abc123", "extra": "data"}', entity_id
+    )
+    assert "Extra keys dict_keys(['extra']) were ignored" in caplog.text
+    chromecast.start_app.assert_called_once_with("abc123")
+    quick_play_mock.assert_not_called()
+
+    # Play_media - media_type cast with unsupported app
+    quick_play_mock.side_effect = NotImplementedError()
+    await common.async_play_media(hass, "cast", '{"app_name": "unknown"}', entity_id)
+    quick_play_mock.assert_called_once_with(ANY, "unknown", {})
+    assert "App unknown not supported" in caplog.text
+
+
+async def test_entity_play_media_sign_URL(hass: HomeAssistantType):
+    """Test playing media."""
+    entity_id = "media_player.speaker"
+
+    await async_process_ha_core_config(
+        hass,
+        {"external_url": "http://example.com:8123"},
+    )
+
+    info = get_fake_chromecast_info()
+
+    chromecast = await async_setup_media_player_cast(hass, info)
+    _, conn_status_cb, _ = get_status_callbacks(chromecast)
+
+    connection_status = MagicMock()
+    connection_status.status = "CONNECTED"
+    conn_status_cb(connection_status)
+    await hass.async_block_till_done()
+
+    # Play_media
+    await common.async_play_media(hass, "audio", "/best.mp3", entity_id)
+    chromecast.media_controller.play_media.assert_called_once_with(ANY, "audio")
+    assert chromecast.media_controller.play_media.call_args[0][0].startswith(
+        "http://example.com:8123/best.mp3?authSig="
+    )
+
+
+async def test_entity_media_content_type(hass: HomeAssistantType):
+    """Test various content types."""
+    entity_id = "media_player.speaker"
+    reg = await hass.helpers.entity_registry.async_get_registry()
+
+    info = get_fake_chromecast_info()
+    full_info = attr.evolve(
+        info, model_name="google home", friendly_name="Speaker", uuid=FakeUUID
+    )
+
+    chromecast = await async_setup_media_player_cast(hass, info)
+    _, conn_status_cb, media_status_cb = get_status_callbacks(chromecast)
+
+    connection_status = MagicMock()
+    connection_status.status = "CONNECTED"
+    conn_status_cb(connection_status)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity_id)
+    assert state is not None
+    assert state.name == "Speaker"
+    assert state.state == "unknown"
+    assert entity_id == reg.async_get_entity_id("media_player", "cast", full_info.uuid)
+
+    media_status = MagicMock(images=None)
+    media_status.media_is_movie = False
+    media_status.media_is_musictrack = False
+    media_status.media_is_tvshow = False
+    media_status_cb(media_status)
+    await hass.async_block_till_done()
+    state = hass.states.get(entity_id)
+    assert state.attributes.get("media_content_type") is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)